build/postgres/Dockerfile

```diff
@@ -9,15 +9,11 @@ ARG POSTGRES_VERSION
 # Update the Ubuntu and PostgreSQL repository indexes
 RUN apt-get update \
     && apt-get install  --no-install-recommends -qy \
-        build-essential \
         ca-certificates \
+        gcc \
         git \
-        libdb-dev \
-        libexpat1-dev \
         libicu-dev \
-        libpq-dev \
-        libxml2-dev \
-        pkg-config \
```

Comment:
`pkg-config` (and probably some other packages) seem to be required to install postgres extensions.

Reply:
I thought we build the postgres extensions in [this image](https://github.com/metabrainz/musicbrainz-docker/blob/mbvm-38-dev/build/postgres/Dockerfile) and they are server side only. Is this correct?

Reply:
Yes but is your link on [_this image_](https://github.com/metabrainz/musicbrainz-docker/blob/mbvm-38-dev/build/postgres/Dockerfile) correct? Because it points to `build/postgres/Dockerfile` this conversation is all about.

Reply:
Yeah, you are right. I thought we where in the other file.

Reply:
I just noticed this very dangerous bug:
````
make: pkg-config: Command not found
cp sql/musicbrainz_collate.sql sql/musicbrainz_collate--1.0.0.sql
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC  -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FORTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o musicbrainz_collate.o musicbrainz_collate.c
make: pkg-config: Command not found
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -shared -o musicbrainz_collate.so musicbrainz_collate.o -L/usr/lib/x86_64-linux-gnu  -Wl,-z,relro -Wl,-z,now  -L/usr/lib/x86_64-linux-gnu/mit-krb5 -Wl,--as-needed
0
````
We don't fail if can't compile. This is the reason why the image built without pkg-config.

Reply:
It seems to compile without the dev libs as strangely the ``.h`` are already there.
The logs are also clean.
````
#6 [3/5] RUN git clone https://github.com/metabrainz/postgresql-musicbrainz...
#6 0.554 Cloning into '/tmp/postgresql-musicbrainz-unaccent'...
#6 1.598 Cloning into '/tmp/postgresql-musicbrainz-collate'...
#6 2.686 Cloning into '/tmp/pg_amqp'...
#6 3.881 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o musicbrainz_unaccent.o musicbrainz_unaccent.c
#6 3.946 musicbrainz_unaccent.c: In function ‘musicbrainz_dunaccentdict_lexize’:
#6 3.946 musicbrainz_unaccent.c:319:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
#6 3.946      char *old_locale = strdup(setlocale(LC_CTYPE, NULL));
#6 3.946      ^~~~
#6 4.018 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -shared -o musicbrainz_unaccent.so musicbrainz_unaccent.o -L/usr/lib/x86_64-linux-gnu  -Wl,-z,relro
 -Wl,-z,now  -L/usr/lib/x86_64-linux-gnu/mit-krb5 -Wl,--as-needed
#6 4.035 cp sql/musicbrainz_unaccent.sql sql/musicbrainz_unaccent--1.0.0.sql
#6 4.090 /bin/mkdir -p '/usr/lib/postgresql/9.5/lib'
#6 4.091 /bin/mkdir -p '/usr/share/postgresql/9.5/extension'
#6 4.092 /bin/mkdir -p '/usr/share/postgresql/9.5/extension'
#6 4.092 /bin/mkdir -p '/usr/share/doc/postgresql-doc-9.5/extension'
#6 4.093 /usr/bin/install -c -m 755  musicbrainz_unaccent.so '/usr/lib/postgresql/9.5/lib/musicbrainz_unaccent.so'
#6 4.098 /usr/bin/install -c -m 644 .//musicbrainz_unaccent.control '/usr/share/postgresql/9.5/extension/'
#6 4.099 /usr/bin/install -c -m 644 .//sql/uninstall_musicbrainz_unaccent.sql .//sql/musicbrainz_unaccent.sql sql/musicbrainz_unaccent--1.0.0.sql '/usr/share/postgresql/9.5/extension/'
#6 4.101 /usr/bin/install -c -m 644 .//README.musicbrainz_unaccent.md '/usr/share/doc/postgresql-doc-9.5/extension/'
#6 4.150 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC  -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FO
RTIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o musicbrainz_collate.o musicbrainz_collate.c
#6 4.233 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -shared -o musicbrainz_collate.so musicbrainz_collate.o -L/usr/lib/x86_64-linux-gnu  -Wl,-z,relro -
Wl,-z,now  -L/usr/lib/x86_64-linux-gnu/mit-krb5 -Wl,--as-needed  -licui18n -licuuc -licudata
#6 4.263 cp sql/musicbrainz_collate.sql sql/musicbrainz_collate--1.0.0.sql
#6 4.306 /bin/mkdir -p '/usr/lib/postgresql/9.5/lib'
#6 4.307 /bin/mkdir -p '/usr/share/postgresql/9.5/extension'
#6 4.309 /bin/mkdir -p '/usr/share/postgresql/9.5/extension'
#6 4.310 /bin/mkdir -p '/usr/share/doc/postgresql-doc-9.5/extension'
#6 4.311 /usr/bin/install -c -m 755  musicbrainz_collate.so '/usr/lib/postgresql/9.5/lib/musicbrainz_collate.so'
#6 4.315 /usr/bin/install -c -m 644 .//musicbrainz_collate.control '/usr/share/postgresql/9.5/extension/'
#6 4.317 /usr/bin/install -c -m 644 .//sql/musicbrainz_collate.sql sql/musicbrainz_collate--1.0.0.sql '/usr/share/postgresql/9.5/extension/'
#6 4.318 /usr/bin/install -c -m 644 .//README.musicbrainz_collate.md '/usr/share/doc/postgresql-doc-9.5/extension/'
#6 4.389 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/pg_amqp.o src/pg_amqp.c
#6 4.478 src/pg_amqp.c: In function ‘amqp_local_phase2’:
#6 4.478 src/pg_amqp.c:99:3: warning: enumeration value ‘XACT_EVENT_PARALLEL_COMMIT’ not handled in switch [-Wswitch]
#6 4.478    switch(event) {
#6 4.478    ^~~~~~
#6 4.478 src/pg_amqp.c:99:3: warning: enumeration value ‘XACT_EVENT_PARALLEL_ABORT’ not handled in switch [-Wswitch]
#6 4.478 src/pg_amqp.c:99:3: warning: enumeration value ‘XACT_EVENT_PRE_COMMIT’ not handled in switch [-Wswitch]
#6 4.478 src/pg_amqp.c:99:3: warning: enumeration value ‘XACT_EVENT_PARALLEL_PRE_COMMIT’ not handled in switch [-Wswitch]
#6 4.478 src/pg_amqp.c:99:3: warning: enumeration value ‘XACT_EVENT_PRE_PREPARE’ not handled in switch [-Wswitch]
#6 4.478 src/pg_amqp.c: In function ‘local_amqp_get_a_bs’:
#6 4.478 src/pg_amqp.c:140:1: warning: type of ‘broker_id’ defaults to ‘int’ [-Wimplicit-int]
#6 4.478  local_amqp_get_a_bs(broker_id) {
#6 4.478  ^~~~~~~~~~~~~~~~~~~
#6 4.478 src/pg_amqp.c: In function ‘local_amqp_get_bs’:
#6 4.478 src/pg_amqp.c:152:1: warning: type of ‘broker_id’ defaults to ‘int’ [-Wimplicit-int]
#6 4.478  local_amqp_get_bs(broker_id) {
#6 4.478  ^~~~~~~~~~~~~~~~~
#6 4.479 src/pg_amqp.c: In function ‘local_amqp_disconnect’:
#6 4.479 src/pg_amqp.c:239:1: warning: type of ‘broker_id’ defaults to ‘int’ [-Wimplicit-int]
#6 4.479  local_amqp_disconnect(broker_id) {
#6 4.479  ^~~~~~~~~~~~~~~~~~~~~
#6 4.599 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_api.o src/librabbitmq/amqp_api.c
#6 4.715 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_connection.o src/librabbitmq/amqp_connection.c
#6 4.833 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_debug.o src/librabbitmq/amqp_debug.c
#6 4.855 src/librabbitmq/amqp_debug.c:44:6: warning: no previous prototype for ‘amqp_dump’ [-Wmissing-prototypes]
#6 4.855  void amqp_dump(void const *buffer, size_t len) {
#6 4.855       ^~~~~~~~~
#6 4.886 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_framing.o src/librabbitmq/amqp_framing.c
#6 5.611 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_mem.o src/librabbitmq/amqp_mem.c
#6 5.668 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_socket.o src/librabbitmq/amqp_socket.c
#6 5.788 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -I. -I./ -I/usr/include/postgresql/9.5/server -I/usr/include/postgresql/internal -Wdate-time -D_FOR
TIFY_SOURCE=2 -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/mit-krb5  -c -o src/librabbitmq/amqp_table.o src/librabbitmq/amqp_table.c
#6 5.871 gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standa
rd -g -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fno-omit-frame-pointer -fPIC -shared -o pg_amqp.so src/pg_amqp.o src/librabbitmq/amqp_api.o src/librabbitmq/amqp_connection.o sr
c/librabbitmq/amqp_debug.o src/librabbitmq/amqp_framing.o src/librabbitmq/amqp_mem.o src/librabbitmq/amqp_socket.o src/librabbitmq/amqp_table.o -L/usr/lib/x86_64-linux-gnu  -Wl,-z,relro -Wl,-z,now  -L
/usr/lib/x86_64-linux-gnu/mit-krb5 -Wl,--as-needed
#6 5.891 cat sql/tables/tables.sql sql/functions/functions.sql > sql/amqp--0.4.2.sql
#6 5.941 /bin/mkdir -p '/usr/lib/postgresql/9.5/lib'
#6 5.942 /bin/mkdir -p '/usr/share/postgresql/9.5/extension'
#6 5.944 /bin/mkdir -p '/usr/share/postgresql/9.5/extension'
#6 5.945 /bin/mkdir -p '/usr/share/doc/postgresql-doc-9.5/extension'
#6 5.946 /usr/bin/install -c -m 755  pg_amqp.so '/usr/lib/postgresql/9.5/lib/pg_amqp.so'
#6 5.951 /usr/bin/install -c -m 644 .//amqp.control '/usr/share/postgresql/9.5/extension/'
#6 5.952 /usr/bin/install -c -m 644 .//updates/amqp--0.4.0--0.4.1.sql .//updates/amqp--0.3.0--0.4.0.sql .//sql/amqp--0.4.2.sql  '/usr/share/postgresql/9.5/extension/'
#6 5.954 /usr/bin/install -c -m 644 .//doc/amqp.md '/usr/share/doc/postgresql-doc-9.5/extension/'
#6 DONE 6.0s
````

Reply:
Issue reported in initial comment fixed in rebased commit 3c49ebaffcb2d7a2e161993660b98f83d35abc24.

Reported Makefile issue as MBS-10670.